x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -175,29 +200,31 @@ func getRegions(svc ec2iface.EC2API) (regionsList []string, err error) {
 	return
 }
 
-func createCloudWatchEvents(getMetricDataOutput *cloudwatch.GetMetricDataOutput, instanceID string, instanceOutput ec2.Instance, regionName string) (event mb.Event, err error) {
-	machineType, err := instanceOutput.InstanceType.MarshalValue()
-	if err != nil {
-		err = errors.Wrap(err, "instance.InstanceType.MarshalValue failed")
-		logp.Error(err)
-	}
-
+func createCloudWatchEvents(getMetricDataOutput *cloudwatch.GetMetricDataOutput, instanceID string, instanceOutput ec2.Instance, regionName string) (event mb.Event, info string, err error) {
 	event.Service = metricsetName
 	event.RootFields = common.MapStr{}
 	mapOfRootFieldsResults := make(map[string]interface{})
```

Comment:
Not related to this PR but perhaps for a later cleanup. I wonder why you use here map[string]interface instead of common.MapStr which is almost the same. Like this you could use below the `Put` method to directly create the correct trees.

Reply:
Yeah originally I was using `Put` method directory but then you don't need the `eventMapping` function and predefined schemas anymore. Seems like all the other modules have schemas in `data.go`, that's why I added it and start using  `mapOfRootFieldsResults := make(map[string]interface{})` here and use `eventMapping` instead of `Put`.